crates/bevy_picking/src/focus.rs

```diff
@@ -190,16 +191,75 @@ fn build_hover_map(
 /// it will be considered hovered.
 #[derive(Component, Copy, Clone, Default, Eq, PartialEq, Debug, Reflect)]
 #[reflect(Component, Default, PartialEq, Debug)]
+#[repr(u8)]
 pub enum PickingInteraction {
     /// The entity is being pressed down by a pointer.
-    Pressed = 2,
+    Pressed(PressedButtons) = 2,
     /// The entity is being hovered by a pointer.
     Hovered = 1,
     /// No pointers are interacting with this entity.
     #[default]
     None = 0,
 }
 
+impl BitOr for PickingInteraction {
+    type Output = Self;
+
+    fn bitor(mut self, rhs: Self) -> PickingInteraction {
+        self |= rhs;
+        self
+    }
+}
+
+impl BitOrAssign for PickingInteraction {
+    fn bitor_assign(&mut self, rhs: Self) {
+        use PickingInteraction::*;
+        match (self, rhs) {
+            (Pressed(a), Pressed(b)) => *a |= b,
+            (Pressed(_), _) => (),
+            (a, Pressed(b)) => *a = Pressed(b),
+            (a @ None, Hovered) => *a = Hovered,
+            _ => (),
+        }
+    }
+}
+
+bitflags::bitflags! {
+    #[repr(transparent)]
+    #[derive(Hash, Clone, Copy, PartialEq, Eq, Debug, Reflect)]
+    #[reflect(opaque)]
+    #[reflect(Hash, PartialEq, Debug)]
+    /// Button pressed in a [`PickingInteraction`]
+    pub struct PressedButtons: u8 {
+        /// Primary mouse button is pressed.
+        const PRIMARY = 1;
+        /// Secondary mouse button is pressed.
+        const SECONDARY = 2;
+        /// Middle mouse button is pressed.
+        const MIDDLE = 4;
+        /// Touch input is pressed.
+        const TOUCH = 8;
```

Comment:
I'm not sure this is correct. Touch inputs are probably akin to a primary press, and some touch devices, like pens, can have a primary secondary, tertiary press, just like a mouse.

Reply:
I've had something like that before: my Note 9 has a pen and it supports hovering over something and clicking.

Reply:
A stylus is not a touch pointer. Winit does not currently support stylus input, so neither do we.

Reply:
It definitely works, I've tested it. It's treated like a touch input.

Regardless, if you are matching on whether a button was pressed, why would you not want it to be considered pressed if it was touched by a different pointing device?

Reply:
The PR just maps the input from picking in a form that makes the most sense to me. Definitely would like to see more detailed info, but that should come from picking first.

Reply:
Renamed to `left` and `right`.